intro-OpenStack/ch_LaunchInstances.tex

```diff
@@ -0,0 +1,272 @@
+  The default security group is assigned to the instance
+  automatically.
+\item[Key Pair] tab. Specify a key pair.
+
+  If the image uses a static root password or a static key set
+  (neither is recommended), you do not need to provide a key pair to
+  launch the instance.
+\item[Configuration] tab. Specify a customization script that runs
+  after your instance launches.
+
+\item[Server Groups] tab.
+
+\item[Scheduler Hints] tab.
+
+\item[Metadata] tab. Add Metadata items to your instance.
+\end{description}
+
+\item Click Launch Instance.
+\end{enumerate}
+
+The instance starts on a compute node in the cloud.
+
+\strong{Note:} If you did not provide a key pair, security groups, or
+rules, users can access the instance only from inside the cloud
+through VNC. Even pinging the instance is not possible without an ICMP
+rule configured.
+
+You can also launch an instance from the Images or Volumes category when
+you launch an instance from an image or a volume respectively.
+
+When you launch an instance from an image, OpenStack creates a local
+copy of the image on the compute node where the instance starts.
+
+For details on creating images, see
+\href{https://docs.openstack.org/image-guide/create-images-manually.html}{\emph{Creating
+images manually}} in the \emph{OpenStack Virtual Machine Image Guide}.
+
+\section{Connect to your instance using SSH}\label{connect-to-your-instance-using-ssh}
+
+\subsection*{Obtain a public ip}
+Only a project's floating ip's can be reached from outside of the
+OpenStack network.  One way to obtain a public ip for an instance, is
+to associate a floating ip to it, as detailed in section
+\ref{allocate-a-floating-ip-address-to-an-instance} on page
+\pageref{allocate-a-floating-ip-address-to-an-instance}.  However,
+only a limited number of floating ip's are available to each cloud
+project, and this approach uses one ip for every instance you want to
+connect to.  A more economical approach, discussed in the next
+section, is to use different ports of a single floating ip to reach
+different instances.
+
+\subsubsection{Floating ip port forwarding}
+OpenStack's networking API, called Neutron, makes it possible to
+forward different ports of the same floating ip to ports in one of
+OpenStack's virtual networks.
+
+You can quickly set up such forwarding rules using
+\lstinline{neutron_port_forward}, a command line tool available on the
+UGent login node, \lstinline{login.hpc.ugent.be}.  In order to use it,
+you must create an application credential for roles ``User'' and
+``heat\_stack\_owner'', and save it as an openrc file (see section
+\ref{sec:appl-cred} on page \pageref{sec:appl-cred}).  Transfer the
+openrc file to your VSC storage space.
+
+\lstinline{neutron_port_forward} can apply various port forwarding
+rules to your instances, by matching the instance names to regular
+expressions.  Use the \lstinline{--help} option to see instructions:
+
+\begin{prompt}
+  % \shellcmd{neutron\_port\_forward ---help}
```

Comment:
@stdweird has included more options in the current `neutron_port_forward`   script. Now it is possible to list the current pf rules, but also it is possible to remove specific rules if that is necessary. This should be documented here.

for instance to list the current available pf rules you can use `--show`     or add ` --id`   to get the rule ID as well.

> $ neutron_port_forward -o $HOME/openrc.sh -m $HOME/portforwarding.ini --show --id
> Server MyServer
> 
> Server MyVMwithNFSshare
>   193.190.85.40:50023 -> 10.10.2.12:22    ID: cda8da24-a24f-4163-a760-7895fa77c35c
> 
> Server myguix
> 
> Server test_heat_instance
> 
> Server testnfsmount
>   193.190.85.40:1000 -> 10.10.2.9:24    ID: cbc1d573-1e21-4fa3-8a90-c5c739d8c403

You can also remove specific PF rules using their IDs like this:

`neutron_port_forward -o $HOME/openrc.sh -m $HOME/portforwarding.ini --remove cbc1d573-1e21-4fa3-8a90-c5c739d8c403`

Reply:
done!